CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEffFromCalibTree.cc

```diff
@@ -229,6 +230,83 @@ SiStripHitEffFromCalibTree::SiStripHitEffFromCalibTree(const edm::ParameterSet&
     nTEClayers = 7;  // number of rings
 
   quality_ = new SiStripQuality(detInfo_);
+
+  layerfound_vsLumi.reserve(::k_END_OF_LAYERS);
+  layertotal_vsLumi.reserve(::k_END_OF_LAYERS);
+  layerfound_vsPU.reserve(::k_END_OF_LAYERS);
+  layertotal_vsPU.reserve(::k_END_OF_LAYERS);
+  layerfound_vsCM.reserve(::k_END_OF_LAYERS);
+  layertotal_vsCM.reserve(::k_END_OF_LAYERS);
+  layerfound_vsBX.reserve(::k_END_OF_LAYERS);
+  layertotal_vsBX.reserve(::k_END_OF_LAYERS);
+}
+
+namespace utils {
+
+  void Recursion(TFile* f1, TDirectory* target) {
+    TString path((char*)strstr(target->GetPath(), ":"));
+    path.Remove(0, 2);
+    f1->cd(path);
+    TDirectory* temp = gDirectory;
+    std::cout << "Checking initial Get Keys Here: " << temp->GetPath() << std::endl;
+
+    TIter next(temp->GetListOfKeys());
+    TKey* key;
+    while ((key = (TKey*)next())) {
+      printf("key: %s points to an object of class: %s \n", key->GetName(), key->GetClassName());
+
+      TObject* obj = key->ReadObj();
+      if (obj->IsA()->InheritsFrom(TDirectory::Class())) {
+        std::cout << "Found subdirectory " << obj->GetName() << std::endl;
+        TDirectory* subdir = (TDirectory*)obj;
+        std::cout << subdir->GetPath() << std::endl;
+        Recursion(f1, subdir);
+      } else if (obj->IsA()->InheritsFrom(TTree::Class())) {
+        std::cout << "object found" << std::endl;
+        delete obj;
+      } else {
+        std::cout << "no new directory:" << std::endl;
+      }
+    }
+  }
+}  // namespace utils
+
+SiStripHitEffFromCalibTree::~SiStripHitEffFromCalibTree() {
+  if (quality_)
+    delete quality_;
+  if (tkmap)
+    delete tkmap;
+  if (tkmapbad)
+    delete tkmapbad;
+  if (tkmapeff)
+    delete tkmapeff;
+  if (tkmapnum)
+    delete tkmapnum;
+  if (tkmapden)
+    delete tkmapden;
+
+  if (fs) {
+    auto& tFile = fs->file();
+    //tFile.Print(); // Print information about the file
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File Name: " << tFile.GetName() << std::endl;
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File Title: " << tFile.GetTitle() << std::endl;
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File Option: " << tFile.GetOption() << std::endl;
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File Writable: " << tFile.IsWritable() << std::endl;
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File IsZombie: " << tFile.IsZombie() << std::endl;
+    edm::LogPrint("") << __PRETTY_FUNCTION__ << " File has inconsistent hash: " << tFile.HasInconsistentHash()
+                      << std::endl;
+
+    bool debug{false};
+
+    if (!tFile.IsZombie() && tFile.IsWritable() && debug) {
+      // Delete all objects in the file recursively
```

Comment:
Why the objects in file are deleted and the files are written and closed only when the `debug` option is `true`?

Reply:
it was an (unsuccessful) attempt to debug https://github.com/cms-sw/cmssw/issues/45084. AFAIU normally the destructor of `TFIleService` should do that automatically, but somewhere there is a double deletion. 
This piece could be removed.